packages/smartgpt-bridge/src/indexerState.ts

```diff
@@ -8,19 +7,19 @@ function baseDir() {
   return path.join(__dirname, "../logs/indexer");
 }
 
-function safeName(s) {
+function safeName(s: string) {
   return String(s || "").replace(/[^a-zA-Z0-9._-]/g, "_");
 }
 
-export function stateDirForRoot(rootPath) {
```

Comment:
we do this indexing thing a few different places, for different data sources. This could probably become it's own service.